packages/agent/src/pid.ts

```diff
@@ -0,0 +1,139 @@
+import { Logger } from '@medplum/core';
+import fs from 'node:fs';
+import { platform, tmpdir } from 'node:os';
+import path from 'node:path';
+import process from 'node:process';
+
+export const pidLogger = new Logger((msg) => `[PID]: ${msg}`);
+
+/**
+ * Get the appropriate PID file location based on OS
+ * @param appName - The name of the application
+ * @returns The path to the PID file
+ */
+export function getPidFilePath(appName: string): string {
+  switch (platform()) {
+    case 'linux':
+    case 'darwin':
+      // We use tmpdir for linux and Mac because /var/run requires root access
+      // The benefit of using the convention is outweighed by the required permission level to create files/directories inside
+      return path.join(tmpdir(), 'medplum-agent', `${appName}.pid`);
+    case 'win32':
+      return path.join('C:', 'ProgramData', 'MedplumAgent', 'pids', `${appName}.pid`);
+    default:
+      throw new Error('Invalid OS');
+  }
+}
+
+/**
+ * Remove the PID file
+ * @param pidFilePath - Path to the PID file
+ */
+export function removePidFile(pidFilePath: string): void {
+  if (fs.existsSync(pidFilePath)) {
+    try {
+      fs.unlinkSync(pidFilePath);
+      pidLogger.info(`PID file removed: ${pidFilePath}`);
+    } catch (err) {
+      pidLogger.error(`Error removing PID file: ${pidFilePath}`, err as Error);
+    }
+  }
+}
+
+/**
+ * Returns true if a PID points to an existing process, else returns false.
+ * @param pid - The PID to check if it exists.
+ * @returns Boolean indicating if a process with the given PID exists.
+ */
+export function checkProcessExists(pid: number): boolean {
+  try {
+    // Sending signal 0 doesn't actually send a signal but checks if process exists
+    process.kill(pid, 0);
+    // If we didn't throw an error above, the process is still running, and PID is not stale
+    return true;
+  } catch (err) {
+    // If the error is "ESRCH", the process doesn't exist
+    if ((err as Error & { code: string }).code === 'ESRCH') {
+      return false;
+    }
+    // If the error is "EPERM", the process exists but we don't have permission
```

Comment:
This is necessary for Windows in the common case of double clicking the exe in `C:\ProgramFiles\MedplumAgent` since it by default will run as a user without permissions to check the service process which is running as an administrator.